hera_cal/lstbin_simple.py

```diff
@@ -0,0 +1,1740 @@
+            antpairs=bl_chunk, 
+            freqs=freq_array, 
+            pols=all_pols,
+            cal_files=cals,
+            time_arrays=time_arrays,
+            time_idx=tinds,
+            ignore_flags=ignore_flags,
+            rephase=rephase,
+            antpos=antpos,
+            lsts=all_lsts,
+            redundantly_averaged=redundantly_averaged,
+            reds=reds,
+            freq_min=freq_min,
+            freq_max=freq_max,
+        )
+
+        slc = slice(nbls_so_far, nbls_so_far + len(bl_chunk))
+
+        if bi == 0:
+            # On the first baseline chunk, create the output file
+            out_files['GOLDEN'] = []
+            for nbin in golden_bins:
+                out_files["GOLDEN"].append(
+                    create_outfile(
+                        kind = 'GOLDEN',
+                        lst=lst_bin_edges[nbin],
+                        times=binned_times[nbin],
+                    )
+                )
+
+            if save_channels and len(binned_times[0]) > 0:
+                out_files['REDUCEDCHAN'] = create_outfile(
+                    kind = 'REDUCEDCHAN',
+                    lst=lst_bin_edges[0],
+                    times=binned_times[0],
+                    channels=list(save_channels),
+                )
+
+        if len(golden_bins)>0:
+            for fl, nbin in zip(out_files['GOLDEN'], golden_bins):
+                write_baseline_slc_to_file(
+                    fl = fl,
+                    slc=slc,
+                    data=data[nbin].transpose((1, 0, 2, 3)),
+                    flags=flags[nbin].transpose((1, 0, 2, 3)),
+                    nsamples=nsamples[nbin].transpose((1, 0, 2, 3)),
+                )
+
+        if "REDUCEDCHAN" in out_files:
+            write_baseline_slc_to_file(
+                fl = out_files['REDUCEDCHAN'],
+                slc=slc,
+                data=data[0][:, :, save_channels].transpose((1, 0, 2, 3)),
+                flags=flags[0][:, :, save_channels].transpose((1, 0, 2, 3)),
+                nsamples=nsamples[0][:, :, save_channels].transpose((1, 0, 2, 3)),
+            )
+
+        data, flags, std, nsamples = reduce_lst_bins(
+            data, flags, nsamples,
+            sigma_clip_thresh = sigma_clip_thresh,
+            sigma_clip_min_N = sigma_clip_min_N,
+            flag_below_min_N=flag_below_min_N,
+        )
+
+        write_baseline_slc_to_file(
+            fl=out_files['LST'],
+            slc=slc,
+            data=data,
+            flags=flags,
+            nsamples=nsamples,
+        )
+        write_baseline_slc_to_file(
+            fl=out_files['STD'],
+            slc=slc,
+            data=std,
+            flags=flags,
+            nsamples=nsamples,
+        )
+
+        nbls_so_far += len(bl_chunk)
+
+    return out_files
+
+
+@profile
+def lst_bin_files(
+    config_file: str | Path,
+    output_file_select: int | Sequence[int] | None=None, 
+    include_autos: bool=True,
+    **kwargs
```

Comment:
I agree, though this is more an issue with `hera_cal`'s lack of sphinx-documentation, I think. I equally don't love copy-pasting massive chunks of docstrings, because they go stale real quick. If we were making sphinx docs that we could look at, it could be dynamically all put together. For now, I've left the kwargs in because I don't want to set the precedent of copy-pasting massive parameter lists. I could probably get some sphinx docs together sometime in the near future.